lib/ipmi_user.c

```diff
@@ -477,16 +481,31 @@ ipmi_user_build_password_prompt(uint8_t user_id)
  *
  * @returns pointer to char with password
  */
-char *
-ask_password(uint8_t user_id)
+int
+ask_password(uint8_t user_id, char *buf, size_t buflen)
 {
-	const char *password_prompt =
-		ipmi_user_build_password_prompt(user_id);
-# ifdef HAVE_GETPASSPHRASE
-	return getpassphrase(password_prompt);
-# else
-	return (char*)getpass(password_prompt);
-# endif
+	const char *password_prompt;
+	char *sbuf;
+	size_t len;
+	int ret = 0;
```

Comment:
It's preferred to default to an error condition rather than success. A success must be explicit.